src/Microsoft.NetCore.Analyzers/Core/Security/SetHttpOnlyForHttpCookie.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Analyzer.Utilities.FlowAnalysis.Analysis.PropertySetAnalysis;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.PointsToAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.NetCore.Analyzers.Security.Helpers;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    class SetHttpOnlyForHttpCookie : DiagnosticAnalyzer
+    {
+        // TODO Lingxia Chen: Help links URLs.
+        internal static DiagnosticDescriptor Rule = SecurityHelpers.CreateDiagnosticDescriptor(
+            "CA5393",
+            typeof(MicrosoftNetCoreAnalyzersResources),
+            nameof(MicrosoftNetCoreAnalyzersResources.SetHttpOnlyForHttpCookie),
+            nameof(MicrosoftNetCoreAnalyzersResources.SetHttpOnlyForHttpCookieMessage),
+            DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            helpLinkUri: null,
+            descriptionResourceStringName: nameof(MicrosoftNetCoreAnalyzersResources.SetHttpOnlyForHttpCookieDescription),
+            customTags: WellKnownDiagnosticTagsExtensions.DataflowAndTelemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create(
+                Rule);
+
+        private static readonly ConstructorMapper ConstructorMapper = new ConstructorMapper(
+            (IMethodSymbol constructorMethod,
+            IReadOnlyList<ValueContentAbstractValue> argumentValueContentAbstractValues,
+            IReadOnlyList<PointsToAbstractValue> argumentPointsToAbstractValues) =>
+            {
+                return PropertySetAbstractValue.GetInstance(PropertySetAbstractValueKind.Flagged);
+            });
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "HttpOnly",
+                (PointsToAbstractValue pointsToAbstractValue) =>
+                   PropertySetAbstractValueKind.Unflagged));
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+                    new HazardousUsageEvaluator(
+                        HazardousUsageEvaluatorKind.Return,
+                        PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown));
```

Comment:
It might be worth considering all the places that HttpCookie can be used instead (or even extend PropertySetAnalysis to evaluate hazardous usages on any argument value), so that we can catch cases like instantiating an HTTP cookie and adding it to an HTTP response.